apm-agent-core/src/test/java/co/elastic/apm/agent/configuration/SpyConfiguration.java

```diff
@@ -51,11 +53,15 @@ public static ConfigurationRegistry createSpyConfig() {
      * <p>
      * That way, the default configuration values are returned but can be overridden by {@link Mockito#doReturn(Object)}
      *
-     * @return a spy configuration registry
      * @param configurationSource
+     * @return a spy configuration registry
      */
     public static ConfigurationRegistry createSpyConfig(ConfigurationSource configurationSource) {
         ConfigurationRegistry.Builder builder = ConfigurationRegistry.builder();
+
+        // log level config from runtime context with highest priority
+        builder.addConfigSource(new TestLogLevelConfigSource());
```

Comment:
[for reviewer] adding this configuration source with high priority allows to ensure the log level is always what we expect, also it replaces the previous value that was set in `test.elasticapm.properties`